ethstorageproof/types.go

```diff
@@ -11,45 +10,42 @@ import (
 	"github.com/ethereum/go-ethereum/common/hexutil"
 )
 
-// BytesHex marshals/unmarshals as a JSON string in hex with 0x prefix.  The empty
-// slice marshals as "0x".
-type BytesHex []byte
+// QuantityBytes marshals/unmarshals as a JSON string in hex with 0x prefix encoded
+// as a QUANTITY.  The empty slice marshals as "0x0".
+type QuantityBytes []byte
 
 // MarshalText implements encoding.TextMarshaler
-func (b BytesHex) MarshalText() ([]byte, error) {
-	result := make([]byte, len(b)*2+2)
-	copy(result, `0x`)
-	hex.Encode(result[2:], b)
-	return result, nil
+func (q QuantityBytes) MarshalText() ([]byte, error) {
+	return []byte(fmt.Sprintf("0x%#x", new(big.Int).SetBytes(q))), nil
```

Comment:
Can you add a comment on this expression? `0x%#x`

Reply:
I've replaced this expression to be more explicit about what's going on.  Actually there was a but, it shouldn't have bee `0x%#x` but `%#x`.  What `%#x` did was format the big.Int in hexadecimal, taking it as an integer (not as a byte slice).  I think it's more clear now, where I use `big.Int.Text(16)`.